### ISSUE TITLE
Bug: [no-useless-default-assignment] false positive on tuple elements at or after a rest element

### ISSUE THREAD

**Issue by Grit03** [typescript-eslint/typescript-eslint#12767]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.
- [x] If I used AI assistance to prepare this issue, I have reviewed and followed the [AI Contribution Policy](https://typescript-eslint.io/contributing/ai-policy/).

### Playground Link

https://typescript-eslint.io/play/#ts=6.0.3&fileType=.tsx&code=CYUwxgNghgTiAEYD2A7AzgF0UgtjqKwaAXPANqYwCWKA5gDTwB0LlNtZAupwNwBQydFjJgcwRrFrwAvPADkMAK4o5nGdjwEi-PqEiwEgzPCoYQOEuRZM2dLo1tSAPvGWgAZjRDBeA1MbJPGGNZOXcoCAgAIygwAGtVdVNzNB4gA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tmUSWWS0AJogBmAQwR1xAygHMmAW0Qt0URNGgdokcGAC%2BIfUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
declare const commands: [string, ...string[]];
const [cmd, arg = 'run'] = commands;

declare const items: [...string[], string | undefined];
const [first = 'fallback'] = items;
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-useless-default-assignment": "error"
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

No errors. Both defaults are reachable at runtime.

- `commands` has a minimum length of 1, so `commands = ['build']` is perfectly valid. There is then nothing at index 1, `arg` is `undefined`, and the `'run'` default kicks in.
- `items` also has a minimum length of 1. With an empty rest element the trailing `string | undefined` shifts into index 0, so `items = [undefined]` is valid. `first` is then `undefined` and the `'fallback'` default is used.

### Actual Result

Both lines are reported, and the autofix deletes the defaults:

- `Default value is useless because the property is not optional. 2:19 - 2:24`
- `Default value is useless because the property is not optional. 5:16 - 5:26`

So running the fixer drops defaults that are actually intended, silently changing runtime behavior.
